### PR TITLE
Move DaemonSet affinities and tolerations to values.yaml to make them user manageable.

### PR DESCRIPTION
**Pull request by FutureMatt**
Moves the CPI DaemonSet affinities and tolerations from daemonset.yaml to values.yaml whilst preserving the defaults.

Some environments, in our case RKEv1, apply different node taints and labels for example, ours are:
```
node-role.kubernetes.io/controlplane=true:NoExecute
node-role.kubernetes.io/etcd=true:NoSchedule
```

This change would allow us to easily adjust them.

### CODE REVIEW

File: charts/vsphere-cpi/values.yaml [hunk 1]
```diff
@@ -77,5 +77,28 @@ daemonset:
...
   ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
   podLabels: {}
-  ## Allows for the default tolerations to be replaced with user-defined ones
-  tolerations: []
+  ## Allows for the default tolerations to be updated or replaced with user-defined ones
+  tolerations:
+    - key: node.cloudprovider.kubernetes.io/uninitialized
+      value: "true"
+      effect: NoSchedule
+    - key: node-role.kubernetes.io/master
```
Comment (XudongLiuHarold) [inline charts/vsphere-cpi/values.yaml:85]: Could you remove this since it is already deprecated?

**Review by XudongLiuHarold — COMMENTED** [review]
Hi @FutureMatt, thanks for your PR!  

Do you have the requirement to not only add some tolerations but also replace them? Cuz now you can already add more tolerations by setting tolerations in `values.yaml`

/ok-to-test

**Comment by FutureMatt** [comment]
> Do you have the requirement to not only add some tolerations but also replace them? Cuz now you can already add more tolerations by setting tolerations in values.yaml

Yes, annoyingly, RKE v1 by default creates control plane nodes with taints and  labels of `controlplane` not `control-plane` and we have ~20 legacy clusters that we now need to apply this to.

**Comment by XudongLiuHarold** [comment]
> > Do you have the requirement to not only add some tolerations but also replace them? Cuz now you can already add more tolerations by setting tolerations in values.yaml
> 
> Yes, annoyingly, RKE v1 by default creates control plane nodes with taints and labels of `controlplane` not `control-plane` and we have ~20 legacy clusters that we now need to apply this to.

make senses to me, but we are about to release 1.31 beta, to make the release stable, I will approve this one after the beta released. :)

**Review by XudongLiuHarold — COMMENTED** [review]
/lgtm

/approve

**Pull request merged**